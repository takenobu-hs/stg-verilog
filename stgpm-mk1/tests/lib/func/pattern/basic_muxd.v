

`define SP1_MUXD

//parameter DW = 32;
  parameter DW = 8;


  /* test pattern */

  initial begin
    @(negedge rst);
    repeat(2) @(posedge clk);

    /* set input data */
    a0 = 'd0;
    a1 = 'd1;
    a2 = 'd2;
    a3 = 'd3;
    a4 = 'd4;
    a5 = 'd5;
    a6 = 'd6;
    a7 = 'd7;

    /* select */
    @(posedge clk) #1 s = 8'b0000_0000;
    @(posedge clk) #1 s = 8'b0000_0001;
    @(posedge clk) #1 s = 8'b0000_0010;
    @(posedge clk) #1 s = 8'b0000_0100;
    @(posedge clk) #1 s = 8'b0000_1000;
    @(posedge clk) #1 s = 8'b0001_0000;
    @(posedge clk) #1 s = 8'b0010_0000;
    @(posedge clk) #1 s = 8'b0100_0000;
    @(posedge clk) #1 s = 8'b1000_0000;
    @(posedge clk) #1 s = 8'bxxxx_xxxx;
    @(posedge clk) #1 s = 8'b0000_0000;
    @(posedge clk) #1 s = 8'b0001_0110;

    repeat(5) @(posedge clk);
    halt();
  end


