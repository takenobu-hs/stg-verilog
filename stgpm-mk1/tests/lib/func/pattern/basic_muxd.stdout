monitor   :       0: xx xx xx xx  xx xx xx xx  xx  xx xx xx xx
monitor   :      70: 00 01 02 03  04 05 06 07  xx  0X 0X 0X 0X
monitor   :      81: 00 01 02 03  04 05 06 07  00  00 00 00 00
monitor   :      91: 00 01 02 03  04 05 06 07  01  00 00 00 00
monitor   :     101: 00 01 02 03  04 05 06 07  02  01 01 01 01
monitor   :     111: 00 01 02 03  04 05 06 07  04  00 02 02 02
monitor   :     121: 00 01 02 03  04 05 06 07  08  00 00 03 03
monitor   :     131: 00 01 02 03  04 05 06 07  10  00 00 00 04
monitor   :     141: 00 01 02 03  04 05 06 07  20  00 00 00 05
monitor   :     151: 00 01 02 03  04 05 06 07  40  00 00 00 06
monitor   :     161: 00 01 02 03  04 05 06 07  80  00 00 00 07
monitor   :     171: 00 01 02 03  04 05 06 07  xx  0X 0X 0X 0X
monitor   :     181: 00 01 02 03  04 05 06 07  00  00 00 00 00
monitor   :     191: 00 01 02 03  04 05 06 07  16  01 03 03 07
